planning/behavior_path_planner/behavior_path_planner_turn_signal-design.md

```diff
@@ -0,0 +1,176 @@
+# Turn Signal Decider
+
+Turn Signal decider determines necessary blinkers.
+
+## Purpose / Role
+
+This module is responsible for activating a necessary blinker during driving. It uses rule-based algorithm to determine blinkers, and the details of this algorithm are described in the following sections. Note that this algorithm is strictly based on the Japanese Road Traffic Raw.
+
+### Limitations
+
+Currently, this algorithm can sometimes give unnatural(not wrong) blinkers in a complicated situations. This is because it tries to follow the road traffic raw and cannot solve `blinker conflicts` clearly in that environment.
```

Comment:
```suggestion
Currently, this algorithm can sometimes give unnatural (not wrong) blinkers in a complicated situations. This is because it tries to follow the road traffic raw and cannot solve `blinker conflicts` clearly in that environment.
```